### DISCUSSION TITLE
How add rc-switch function on tasmota32-bluetooth firmware?

### DISCUSSION THREAD

**Discussion by progman27** [arendst/Tasmota#13285] [Q&A] [answered]
Hi, how add rc-switch function on tasmota32-bluetooth firmware? i'm add #define USE_RC_SWITCH string, but compiling failed. I'm use gitpod service for compiling. Thank you.

**Comment by sfromis** [comment, marked as answer]
The general concept is that the standard build set up does not allow  you to add features to the prebaked tasmota32-xxxxx build environments, but you can add both to the base "tasmota32". However, IIRC there are special requirements for BLE, so you may have to create a [custom firmware](https://tasmota.github.io/docs/Compile-your-build/#defining-multiple-custom-firmwares) build environment.

**Reply by barbudor** [reply]
@meloman27
To compile `tasmota32` with Bluetooth you need to have `platformio_override.ini` and uncomment one of the additional libraries
I'm not on my PC right now but you should be able to easily the line as there is a clear comment before about that

**Reply by progman27** [reply]
I'm uncomment tasmota32 and compile tasmota32 with command: platformio run -e tasmota32
Result: /workspace/Tasmota/tasmota/xdrv_79_esp32_ble.ino:148:26: fatal error: NimBLEDevice.h: No such file or directory. compilation terminated.

**Reply by barbudor** [reply]
So you didn't uncomment the right line 😂
```
; *** uncomment the following line if you use Bluetooth or Apple Homekit in a Tasmota32 build. Reduces compile time
;                          lib/libesp32_div
```
Line 131

**Reply by progman27** [reply]
Wow! You are a genius! Everything worked out, thank you very much!!!

**Reply by barbudor** [reply]
I just went through the same question 2 days ago  🤣

**Reply by progman27** [reply]
Cool! :)
